cmip6_downscaling/methods/deepsd/tasks.py

```diff
@@ -0,0 +1,475 @@
+    shifted_ds = bilinear_interpolate(ds=orig_ds, output_degree=output_degree)
+    shifted_ds.attrs.update({'title': 'shift'}, **get_cf_global_attrs(version=version))
+    print(f'writing shifted dataset to {target}')
+    shifted_ds.pipe(set_zarr_encoding).to_zarr(target, mode='w')
+    return target
+
+
+@task(log_stdout=True)
+def coarsen_obs(path: UPath, output_degree: float) -> UPath:
+    """Coarsen grid using conservative interpolation.
+
+    Parameters
+    ----------
+    path : UPath
+        Path to original (likely observational) dataset.
+    output_degree : float
+        Resolution to coarsen the dataset to.
+
+    Returns
+    -------
+    UPath
+        Path to coarsened dataset.
+    """
+    # Similar to coarsen_and_interpolate in GARD tasks (maybe could be combined?)
+    ds_hash = str_to_hash(str(path) + str(output_degree))
+    target = intermediate_dir / 'coarsen' / ds_hash
+
+    if use_cache and zmetadata_exists(target):
+        print(f'found existing target: {target}')
+        coarse_ds = xr.open_zarr(target)
+        return target
+
+    orig_ds = xr.open_zarr(path)
+
+    # Coarsen obs
+    coarse_ds = conservative_interpolate(ds=orig_ds, output_degree=output_degree)
+    coarse_ds.attrs.update({'title': 'coarsen_interpolate'}, **get_cf_global_attrs(version=version))
+    print(f'writing coarsened dataset to {target}')
+    coarse_ds.pipe(set_zarr_encoding).to_zarr(target, mode='w')
+    return target
+
+
+@task(log_stdout=True)
+def coarsen_and_interpolate(path: UPath, output_degree: float) -> UPath:
+    """Coarsen grid and interpolate back to twice the coarsened grid resolution.
+
+    Parameters
+    ----------
+    path : UPath
+        Path to original (likely observational) dataset.
+    output_degree : float
+        Resolution to coarsen the dataset to.
+
+    Returns
+    -------
+    UPath
+        Path to interpolated dataset with a resolution twice that specified by
+        ``output_degre``.
+    """
+    # Similar to coarsen_and_interpolate in GARD tasks (maybe could be combined?)
+    ds_hash = str_to_hash(str(path) + str(output_degree))
+    target = intermediate_dir / 'coarsen_interpolate' / ds_hash
+
+    if use_cache and zmetadata_exists(target):
+        print(f'found existing target: {target}')
+        interpolated_ds = xr.open_zarr(target)
+        return target
+
+    orig_ds = xr.open_zarr(path)
+
+    # Coarsen obs
+    coarse_ds = conservative_interpolate(ds=orig_ds, output_degree=output_degree)
+    # Interpolate back to 2x higher resolution
+    interpolated_ds = bilinear_interpolate(ds=coarse_ds, output_degree=output_degree / 2)
+    interpolated_ds.attrs.update(
+        {'title': 'coarsen_interpolate'}, **get_cf_global_attrs(version=version)
+    )
+    print(f'writing interpolated dataset to {target}')
+    interpolated_ds.pipe(set_zarr_encoding).to_zarr(target, mode='w')
```

Comment:
Same suggestion as in https://github.com/carbonplan/cmip6-downscaling/pull/239/files#r906733871